tests/python/test_short_circuit.py

```diff
@@ -0,0 +1,49 @@
+import taichi as ti
+
+
+@ti.test(short_circuit_operators=True)
+def test_and_shorted():
+    a = ti.field(ti.i32, shape=10)
+
+    @ti.func
+    def explode() -> ti.i32:
+        return a[-1]
```

Comment:
To actually make this explode, you need to add `debug=True`. When `debug=False`, indexing out of bounds is an undefined behavior in Taichi, which may or may not trigger a runtime error.